include/color.h

```diff
@@ -51,4 +51,24 @@ typedef union {
     u16 rgba;
 } Color_RGBA16;
 
+typedef union {
+    struct {
+        u32 r : 5;
+        u32 g : 5;
+        u32 b : 5;
+        u32 a : 1;
+    };
+    u16 rgba;
+} Color_RGBA16_2;
```

Comment:
What's the difference between this and `Color_RGBA16`? Can't we merge those two?

Reply:
Because division is involved, the r,g,b needs to either be typed as or cast to u32 in sys_cmpdma to generate the correct codegen in func_80178AC0, otherwise there will be extra checks introduced. 

Color_RGBA16 is only used in PreRender_AntiAliasAlgorithm, and changing the type from u16 -> u32 mostly works except that the pxIn right change to logical instead of arithmetic, and require an s32 cast to fix this.

So yes they can be combined, but we would need to introduce a cast either on the sys_cmpdma side or PreRender_AntiAliasAlgorithm side.

Reply:
Is the one file/function that this type is used in used anywhere outside of it's own file? Might make sense to put this struct in the C file where it's used, unless the function is public.